generator/src/test/java/com/linkedin/pegasus/generator/TestPegasusDataTemplateGenerator.java

```diff
@@ -205,4 +206,62 @@ private void testRunGenerator(String pegasusFilename, Map<String, String> expect
             file -> file.getName().replace(".java", ""),
             Function.identity()));
   }
+
+  @DataProvider(name = "test_schema_permutation_deterministic")
+  private Object[][] createPermutedDataTemplateCases()
+  {
+    return new Object[][]
+        {
+            {"IsoDuration.pdsc", "PremiumService.pdsc", "PremiumService.pdsc", "IsoDuration.pdsc"},
+            {"AField.pdl", "ARecord.pdl", "ARecord.pdl", "AField.pdl"},
+            {"BRecord.pdl", "BField.pdl", "BField.pdl", "BRecord.pdl"},
+        };
+  }
+
+  @Test(dataProvider = "test_schema_permutation_deterministic")
+  public void testDataTemplateGenerationDeterminism(String[] testArgs)
+      throws Exception
+  {
+    int permuteLength = testArgs.length / 2;
+    String[] pegasusFilenames1 = Arrays.copyOfRange(testArgs, 0, permuteLength);
+    String[] pegasusFilenames2 = Arrays.copyOfRange(testArgs, permuteLength, testArgs.length);
+    File[] generatedFiles1 = generateDataTemplateFiles(pegasusFilenames1);
+    File[] generatedFiles2 = generateDataTemplateFiles(pegasusFilenames2);
+    checkGeneratedFilesConsistency(generatedFiles1, generatedFiles2);
+  }
+
+  private File[] generateDataTemplateFiles(String[] pegasusFilenames) throws Exception
+  {
+    File tempDir = Files.createTempDirectory("restli").toFile();
+    File argFile = new File(tempDir, "resolverPath");
+    Files.write(argFile.toPath(), Collections.singletonList(pegasusDir));
+    System.setProperty(AbstractGenerator.GENERATOR_RESOLVER_PATH, String.format("@%s", argFile.toPath()));
+    String tempDirectoryPath1 = _tempDir.getAbsolutePath();
+    String[] mainArgs = new String[pegasusFilenames.length + 1];
+    mainArgs[0] = tempDirectoryPath1;
+    for (int i = 0; i < pegasusFilenames.length; i++)
+    {
+      mainArgs[i+1] = new File(pegasusDir + FS + pegasusFilenames[i]).getAbsolutePath();
+    }
+    PegasusDataTemplateGenerator.main(mainArgs);
+    File[] generatedFiles = _tempDir.listFiles((File dir, String name) -> name.endsWith(".java"));
```

Comment:
Is this guaranteed to return the files sorted alphabetically?

Reply:
the returning files are not needed to be sorted, (if it sorts that is fine), the requirement is that for two consecutive calls of `_tempDir.listFiles` , the  returning files order shall be the same.

Reply:
I don't see that consecutive calls would return the same order: https://docs.oracle.com/javase/7/docs/api/java/io/File.html#listFiles()

I suggest sorting the generateFiles array to be sure this cannot cause flakiness.